src/platforms/rp2040/src/lib/gpiodriver.c

```diff
@@ -122,6 +122,7 @@ static term nif_gpio_set_pin_pull(Context *ctx, int argc, term argv[])
     if (UNLIKELY(gpio_num >= NUM_BANK0_GPIOS)) {
         RAISE_ERROR(BADARG_ATOM);
     }
+    VALIDATE_VALUE(argv[1], term_is_atom);
     int pull_mode = interop_atom_term_select_int(pull_mode_table, argv[1], ctx->global);
```

Comment:
I think there is some kind of API design fallacy here, since it fails if an unexpected term type is provided, but it defaults to floating if an unrelated atom is given.
I think it should fail if any unexpected value is provided (of any kind).

Reply:
I completely agree with you. This is also the same behavior exhibited by the esp32 gpio driver.

Reply:
I will fix this here. And fix the behavior of the esp32 gpio driver in a separate PR to match.

Reply:
This is fixed. Now any invalid pull mode, bad atom or otherwise will result in `badarg` being raised.

Reply:
interop_atom_term_select_int will select for you `-1` if something that is not an atom is provided, making `VALIDATE_VALUE(argv[1], term_is_atom);` check redundant.

Reply:
this is fixed.